### ISSUE TITLE
Get stable network with TomP2P (port forwarding or relay) including working with sendDirect

### ISSUE THREAD

**Issue by ManfredKarrer** [bisq-network/bisq#230]
Related to #3.
There are 2 problems when port forwarding is used:
- sendDirect does not work
- future.isSuccess() is false but actually the DHT operation has succeeded.

At the moment port forwarding is deactivated and the future.isSuccess() failure case is also commented out. We need to wait until Thomas Bocek has fixed those problems, or start to investigate deeper the TomP2P code.

**Comment by notmandatory** [comment]
I created #232 as a proposed redesign of the trade process to use only the DHT and no direct communications.

**Comment by notmandatory** [comment]
Will focus on unit/integration type tests to confirm latest TomP2P versions are working or not.

**Comment by ManfredKarrer** [comment]
Update:
thomas told me that in port forwarding mode the sendDirect will not work because NAT reflection is not supported by most routers.
so that means that my tests have failed because it is not possible (by most routers) that you test 2 clients connecting to each other and both are behind NAT. that is no problem for real usage, just for testing now.
i will try to get a 2. internet access.
so if we have luck all works already good in port forwarding mode when you just connect 1 client.
we tried also that he get the app running so he can test the open problems (replay mode) with our app directly, but there are strange problems, i assume from his environment (he use openjdk with a manual compiled jfx and arch linux).

I wanted to test last night with oakpacific (mint linux) but he got some errors and as it was in chinese and he did not manage to get IDEA wiht english started, we gave up. I have seen also in my local tests that the current TomP2P codebase still has some issues. I will try to fix those for the hangout session so we should be able to test it.
